operator/pkg/translate/translate.go

```diff
@@ -243,13 +246,150 @@ func (t *Translator) OverlayK8sSettings(yml string, iop *v1alpha1.IstioOperatorS
 		if err != nil {
 			return "", err
 		}
+
+		// Apply the workaround for merging service ports with (port,protocol) composite
+		// keys instead of just the merging by port.
+		if strings.HasSuffix(inPath, "Service") {
```

Comment:
better to do an exact match on the last path element here.

Reply:
Done.